protocol/eth-shared-lockbox.md

```diff
@@ -0,0 +1,120 @@
+
+### Chain Management
+
+This refers to how OP Chains will be governed within an interoperable set of chains (also called a "cluster"). Once governance principles are well-established from the perspective of the technological stack, state transitions can be secured, enabling ETH to be uniformly integrated and secured across the entire cluster. Currently, this is achieved by addressing the following:
+
+- The interoperable set of chains will be governed by an entity responsible for:
+    - Approving chains to join the cluster. This is done either through trusted deployment methods (e.g., OP Stack Manager) or by approval after security checks are performed.
+    - Managing (adding and removing) chains within the interoperable graph.
+    - Approving protocol upgrades for all involved.
+    - Taking action if an OP Chain decides not to follow the established standard (e.g. Expresses a desire to leave the cluster.).
+- Each chain is capable of defending against any maliciously claimed state transition. Ideally, this is accomplished through a set of security features at the proof level, such as permissionless proofs, capability of proving executions that are related to interop executions, and the presence of the Guardian role for the whole system. This is indifferent to what proof system is used (ZK, fault proofs, etc.).
+- Joining a cluster will automatically enable the use of `SuperchainWETH` for the specific set of chains defined by the `dependencyManager`.
+
+### SuperchainWETH usage
+
+As mentioned above, in any OP Chain within the cluster where interop is enabled, the use of `SuperchainWETH` will be activated. As a result, equivalence between ETH deposits and the actual ETH supply will vary from the outset. In a world with freedom of movement, all ETH liquidity is theoretically shared across the entire cluster eventually, regardless of how deposits are handled. This is an unavoidable scenario that deserves careful consideration and is crucial for understanding the following sections.
+
+# Solution
+
+We propose an L1 shared liquidity design, achieved by introducing a new `SharedLockbox` in L1 that serves as a singleton ETH contract for the entire interoperable set of chains. New ETH deposits will be forwarded to the lockbox, and the same applies to ETH withdrawals. This could also serve as the singleton contract for gas tokens in the case of CGT chains.
+
+### Spec changes
+
+The core changes proposed are the following:
+
+- Introduce `SharedLockbox` which accepts deposits and withdrawals of ETH and Gas Tokens as well.
+- Modify `OptimismPortal` to forward ETH deposits into `SharedLockbox` and allow extraction of ETH when a withdrawal is finalized.
+- Maintain an on-chain list (through `SystemConfigInterop` or another contract if already defined) that registers the chains that are part of the cluster. This allows `SharedLockbox` to validate a withdrawal from a chain that is part of the cluster.
```

Comment:
This control mechanism is really important. I would love to enumerate some possible designs. Right now there is a "dependency set manager" role in the `SystemConfig`. It was originally thought that this role would be governance/security council, but now maybe the default config should define it as this contract and then this contract is owned by gov/security council and it holds the aggregate dependency set and calls each individual chain's `SystemConfig` to manage the dependency set. Or we should rethink how the dependencies are managed from first principles, another thing that I realized recently is that the dependencies are not maintained within the `SystemConfig` on L1, they only exist at the moment on L2 in the `L1Block` contract.

Reply:
I have taken this feedback and incorporated a new contract called `DependencySetManager` to manage the `dependencyManager` role.